addons/website_event_track/static/src/js/event_track_timer.js

```diff
@@ -0,0 +1,113 @@
+odoo.define('website_event_track.event_track_timer', function (require) {
+
+'use strict';
+
+const publicWidget = require('web.public.widget');
+
+const Timer = publicWidget.Widget.extend({
+
+    //--------------------------------------------------------------------------
+    // Widget
+    //--------------------------------------------------------------------------
+
+    /**
+     * @override
+     * @param {{ref: moment, ttl: Integer, ?delay: Integer}}
+     *  ref - Time of reference
+     *  ttl - Time to live (in seconds)
+     *  delay - Minimum delay between two updates (in ms)
+     */
+    init: function (parent, params) {
+        this._super.apply(this, arguments);
+        this.delay = params.delay || 1000;
+        this.deadline = params.ref.add(params.ttl, 'seconds');
+    },
+
+    /**
+     * @override
+     */
+    start: function () {
+        return this._super.apply(this, arguments).then(() => {
+            let dt = this.deadline.diff(moment());
+            if (dt > 0) {
+                this.trigger_up('init', { dt });
+                setInterval(this._update.bind(this), this.delay);
+            } else {
+                this.trigger_up('time_up');
+            }
+        });
+    },
+
+    /**
+     * @override
+     */
+    destroy: function () {
+        clearInterval(this.interval);
+        this._super(...arguments);
+    },
+
+    // -------------------------------------------------------------------------
+    // Private
+    // -------------------------------------------------------------------------
+
+    _update: function () {
+        let dt = this.deadline.diff(moment());
+        if (dt > 0) {
+            this.trigger_up('update', { dt });
+        } else {
+            clearInterval(this.interval);
+            this.trigger_up('time_up');
+        }
+    }
+});
+
+publicWidget.registry.EventTrackTimer = publicWidget.Widget.extend({
```

Comment:
I deliberately separated the logic into two components so that the timer could be reused somewhere else. If you think that this widget will never be reused, I can merge these two components :)